airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationDefinitionsHandler.java

```diff
@@ -84,7 +84,7 @@ static DestinationDefinitionRead buildDestinationDefinitionRead(final StandardDe
           .name(standardDestinationDefinition.getName())
           .dockerRepository(standardDestinationDefinition.getDockerRepository())
           .dockerImageTag(standardDestinationDefinition.getDockerImageTag())
-          .documentationUrl(new URI(standardDestinationDefinition.getDocumentationUrl()))
+          .documentationUrl(Uris.fromStringOrNull(standardDestinationDefinition.getDocumentationUrl()).orElse(null))
```

Comment:
@colesnodgrass you mentioned this being a bad pattern. what would you suggest instead?

Reply:
The `orElse` usage here is fine as no object will be allocated with the `.orElse(null)` call.  If there was an additional object being created  in the _else_ case, then we should `orElseGet` instead.

On a related node, the method `fromStringOrNull` could be simplified from
```
  public static Optional<URI> fromStringOrNull(@Nullable final String stringOrNull) throws URISyntaxException {
    if (stringOrNull == null) {
      return Optional.empty();
    } else {
      return Optional.of(new URI(stringOrNull));
    }
  }
  ```
  to 
  ```
  public static Optional<URI> fromStringOrNull(@Nullable final String stringOrNull) throws URISyntaxException {
    return Optional.ofNullable(stringOrNull).map(s -> new URI(s));
  }
  ```
 ---
After some further analysis of this PR, this method shouldn't return `Optional` at all.  Every usage of it is immediately followed up by `.orElse(null)`.  Easier to have this method return a `null` if the provided string was a `null` and document that behavior on the method and in the unit-test for it.
```
  public static URI fromStringOrNull(@Nullable final String stringOrNull) throws URISyntaxException {
    if (stringOrNull == null) {
      return null;
    } 
    return new URI(stringOrNull);
  }
```

Reply:
Just for fun, If we were writing this in kotlin, we could simplify this even more with the [let scope function](https://kotlinlang.org/docs/scope-functions.html#let):
```
.documentationUrl(standardDestinationDefinition.getDocumentationUrl()?.let{ URI(it) })
```

Reply:
```
public static Optional<URI> fromStringOrNull(@Nullable final String stringOrNull) throws URISyntaxException {
  return Optional.ofNullable(stringOrNull).map(s -> new URI(s));
}
```
this doesn't work because `new URI()` returns a non-runtime exception so it can't go in `map` easily and propagate the exception.

Reply:
Doh, you're correct. Didn't realize it was a checked exception.